code/modules/atmospherics/machinery/components/unary_devices/tank.dm

```diff
@@ -1,83 +1,356 @@
-#define AIR_CONTENTS ((25*ONE_ATMOSPHERE)*(air_contents.volume)/(R_IDEAL_GAS_EQUATION*air_contents.temperature))
 /obj/machinery/atmospherics/components/unary/tank
-	icon = 'icons/obj/atmospherics/pipes/pressure_tank.dmi'
-	icon_state = "generic"
+	icon = 'icons/obj/atmospherics/stationary_canisters.dmi'
+	icon_state = "smooth"
 
 	name = "pressure tank"
 	desc = "A large vessel containing pressurized gas."
 
 	max_integrity = 800
+	integrity_failure = 0.2
 	density = TRUE
 	layer = ABOVE_WINDOW_LAYER
+
+	custom_materials = list(/datum/material/alloy/plasteel=4000, /datum/material/iron=20000)
+	material_flags = MATERIAL_GREYSCALE | MATERIAL_ADD_PREFIX | MATERIAL_AFFECT_STATISTICS
+
 	pipe_flags = PIPING_ONE_PER_TURF
+	device_type = QUATERNARY
```

Comment:
Why you owerride unary device instead of transferring it to quarternary device type?

Reply:
Because this all still a draft pr and I don't want to have to edit every map yet